flocker/node/agents/blockdevice.py

```diff
@@ -803,11 +803,17 @@ def discover_state(self, node_state):
         """
         volumes = self.block_device_api.list_volumes()
 
-        manifestations = {
-            m.dataset_id: m for m in (
-                _manifestation_from_volume(v) for v in volumes
-                if v.host == self.hostname)
-        }
+        manifestations = {}
+        nonmanifest = {}
+
+        for volume in volumes:
+            dataset_id = unicode(volume.dataset_id)
+            if volume.host == self.hostname:
+                manifestations[dataset_id] = _manifestation_from_volume(
```

Comment:
Will unmounted datasets be handled in a separate issue? If so add a comment/link, otherwise maye it should be done here?